### PR TITLE
fetch asset id from app params and add app hash check

### PR DESCRIPTION
**Pull request by rmeena840**

### CODE REVIEW

File: examples/dao/setup_sigma_dao.sql [hunk 1]
```diff
@@ -19,7 +19,8 @@ RETURNS TRIGGER
 AS $$
 DECLARE
     asset_id bigint;
-	log_sigma text := 'U2lnbWFEQU8gY3JlYXRlZA==';
+	log_sigma text := 'U2lnbWFEQU8gY3JlYXRlZA=='; -- Byte code of 'SigmaDAO created'
+	gov_asset text : ='Z292X3Rva2VuX2lk'; -- Byte code of 'gov_token_id'
```
Comment (vuvoth) [inline examples/dao/setup_sigma_dao.sql:23]:
```suggestion
	gov_asset text := 'Z292X3Rva2VuX2lk'; -- Byte code of 'gov_token_id'
```

File: examples/dao/setup_sigma_dao.sql [hunk 2]
```diff
@@ -32,9 +33,8 @@ BEGIN
 		/*	sigma dao contrac check. Here 'dt' and 'lg' is a json object present
 			in each record of txn column */
 		IF i.txn::jsonb -> 'dt'->'lg' ?& ARRAY[log_sigma] THEN
-			-- TODO: Fetch asset id from global state not from latest entry in asset table
-			-- store token id from db to above decalred variable asset_id
-			SELECT index INTO asset_id FROM public.asset ORDER BY index DESC LIMIT 1;
+			-- Iterate json object and get gov asset from it
+			SELECT i.txn::jsonb -> 'dt'->'gd'->gov_asset->'ui' INTO asset_id;
```
[the remark this answers is not in the document]
Comment (rmeena840) [reply examples/dao/setup_sigma_dao.sql:46]: Do you mean it does not exist??
If it does not exist then asset_id will be set with null.

[the remark this answers is not in the document]
Comment (vuvoth) [reply examples/dao/setup_sigma_dao.sql:44]: Ah, Yes! Doesn't exist. I think it should not be null! I think we should modify smart contract to check it.

File: examples/dao/setup_sigma_dao.sql [hunk 3]
```diff
@@ -18,27 +18,34 @@ CREATE OR REPLACE FUNCTION sigmadao_trigger_fn()
...
...cEYg0QIhJAAL4yByIrYg4yByIoYg0QMgciKGINIiliJwZkDxAiKWIiJwRiDRAjEhBAAAEAJTUANAAiEiInCWIjEhBEIicFYiISQABSIicFYiUSQAAXIicFYiEEEkAAAQAyBCISRCInCSJmIkMyBCUSMwEQJBIQMwETIicKYhIQMwEUIicLYhIQMwESIicMYhIQMwERIicSYhIQREL/xzIEJRIzARAiEhAzAQAiJwpiEhAzAQciJwtiEhAzAQgiJwxiEhBEQv+eIjUAQv9qIQQ1AEL/YyQ1AEL/XSMjJw02HAFQYzUHNQYyBCISIicRYjIHDhAyByIoYg4QIypiIw0QRDQHIxJAAHc0BiInB2ITQABeADcAGgEpEkAARzcAGgEnBBJAAC43ABoBJw4SQAABACInDiInDmIjKmIIZiMnE2IiKGIOQAACIkMjJxMiKGJmQv/0IicEIicEYiMqYghmQv/aIikiKWIjKmIIZkL/zSMnDTYcAVAiJwdiZkL/lCMnDTYcAVAiJwdiZkL/hTIEIg8iOBAkEhAiOBEnCGQSECI4FDIKEhAiOBIjDxBEIyojKmIzARIIZiJDIycFYiMSRDIEIg8iOBAkEhAiOBEnCGQSECI4FDIKEhAiOBIjDxBEMwEgMgMSMwESKmQSEEQjJxQ2GgFmIycPNhoCZiMnFTYaA2Y2GgSAABJAANYjJxA2GgRmNhoFFzUCNhoGFzUDNhoHFzUENhoIFzUFNAIyBw1EIycRNAJmNAM0Ag0nF2Q0AzQCCQ4QJxhkNAM0AgkPEEQjKDQDZjQENAMNRCMrNARmNAUiEjQFJRIRNAUhBBIRRCMnBTQFZiMnBWIiEkAASSMnBWIlEkAAHyMnBWIhBBJAAAEAIycWNhoJZiMnBzEXZiMnCSNmIkMjJwo2GglmIycSNhoKF2YjJws2GgtmIycMNhoMF2ZC/9IjJwo2GglmIycLNhoKZiMnDDYaCxdmQv+5IycQgAZzaGEyNTZmQv8iMgQiEjEgMgMSEDYwACcIZBIQRLEkshAnCGSyETIKshQjshIjsgGzIkMyBCISRCJDI0M2GgIXIw02GgIXNhoDFwwQRCo2GgAXZycGNhoBF2cnFzYaAhdnJxg2GgMXZycPNhoEZ4AIZGFvX25hbWU2GgVnJwg2GgYXZ4AQU2lnbWFEQU8gY3JlYXRlZLAiQw==';
```
[the remark this answers is not in the document]
Comment (rmeena840) [reply examples/dao/setup_sigma_dao.sql:24]: ok. I would suggest `sigma_dao_app_bytecode` and `app_bytecode`. Since bytecode is specific to sigma dao.

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**